### PR TITLE
Hide mods page in devloper mode

### PR DESCRIPTION
**Pull request by artehe**
When you enable developer mode make the mods tab un-navigatable

### CODE REVIEW

**Comment by mihaicm93** [comment]
I have no idea if it would make sense to add it to this PR or make another but would be good to have a check for the plugins folder if there is any other .dll's except stayintarkov.dll and configmanager.dll to get an error msg or just remove the extra mods

**Comment by artehe** [comment]
> I have no idea if it would make sense to add it to this PR or make another but would be good to have a check for the plugins folder if there is any other .dll's except stayintarkov.dll and configmanager.dll to get an error msg or just remove the extra mods

I was planning on doing this at some point, the part I can't decide on is where to put it at the moment. 

I'm leaning towards running the check itself on when you enable developer mode have it yell at you if there's any mods it detects as installed and tells you to get rid of them first.

**Comment by Plootie** [comment]
> I'm leaning towards running the check itself on when you enable developer mode have it yell at you if there's any mods it detects as installed and tells you to get rid of them first.

IMO this is the best solution

**Comment by artehe** [comment]
> IMO this is the best solution

Cool, I'll do that then.

It'll be reliant on #158 as I can use the updated mod service to check if they have mods we don't mind them using while testing SIT builds such as configuration manager. 

We can also add the same check onto the connect button or whatever it'll be in the play page as well, but that can wait till @Plootie has finished with the UI changes there. Just to really make sure users can accidentally do something we don't want them to.

**Comment by artehe** [comment]
@mihaicm93 @Plootie 

I think this is working happily now feel free to test it and see what you think should have the additional check to stop users enabling developer mode with any mods apart from Configuration manager installed.

**Comment by Plootie** [comment]
👍I'll give this a test soon :3

**Comment by mihaicm93** [comment]
Works great, I will have to get a Pre-release to try it fully with that but otherwise all good on my end
👍

**Review by mihaicm93 — APPROVED** [review]

File: SIT.Manager/ViewModels/MainViewModel.cs [hunk 1]
```diff
@@ -70,7 +73,14 @@ public partial class MainViewModel : ObservableRecipient, IRecipient<Installatio
...
-        _managerConfigService.ConfigChanged += async (o, c) => await CheckForUpdate();
+        _managerConfigService.ConfigChanged += ManagerConfigService_ConfigChanged;
+    }
+
+    private async void ManagerConfigService_ConfigChanged(object? sender, ManagerConfig e)
+    {
+        IsDevloperModeEnabled = e.EnableDeveloperMode;
+
+        await CheckForUpdate();
     }
```
Comment (Plootie) [inline SIT.Manager/ViewModels/MainViewModel.cs:84]: Maybe silly question. Why do we check for an update when toggling dev mode and if this is needed will caching get in the way of this working properly?

**Review by Plootie — COMMENTED** [review]
Just need to perform some tests with this but the code looks good. Just this one question

**Comment by Plootie** [comment]
Found a pretty hilarious bug that I don't even know how to approach honestly. When enabling the developer mode we hide the mods page, however if you navigate to the mod page first, then enable the dev mode and use the "browser back" buttons on your mouse you can navigate back to the mods page while in dev mode. Idk how many people would ever actually do this but might be worth trying to stop

**Comment by mihaicm93** [comment]
That is crazy, so I guess a check before game on the plugins folder?

**Comment by artehe** [comment]
> Found a pretty hilarious bug that I don't even know how to approach honestly. When enabling the developer mode we hide the mods page, however if you navigate to the mod page first, then enable the dev mode and use the "browser back" buttons on your mouse you can navigate back to the mods page while in dev mode. Idk how many people would ever actually do this but might be worth trying to stop

Well that's unexpected. I think I found the cause those

File: SIT.Manager/ViewModels/MainViewModel.cs (hunk 1 above)
Comment (artehe) [reply SIT.Manager/ViewModels/MainViewModel.cs:84]: That's been there the whole time, I just stuck ```IsDevloperModeEnabled = e.EnableDeveloperMode;``` in there as that was the most appropriate place for it.

probably should look into that separately as it does currently run the app update check every time that there is a ManagerConfig change. It does also check for a SIT update but that only happens once every hour

**Pull request merged**